lib/grpc/stub.ex

```diff
@@ -150,11 +150,18 @@ defmodule GRPC.Stub do
     parse_unary_response(response, unmarshal)
   end
 
-  defp parse_unary_response({_headers, data_list}, unmarshal) do
-    data_list
-    |> Enum.join("")
-    |> GRPC.Message.from_data
-    |> unmarshal.()
+  defp parse_unary_response({headers, data_list}, unmarshal) do
+    {_, status} = Enum.find(headers, nil, fn(header) -> elem(header, 0) == "grpc-status" end)
+    if status != GRPC.Status.ok do
+      {_, message} = Enum.find(headers, nil, fn(header) -> elem(header, 0) == "grpc-message" end)
+      {:error, %GRPC.RPCError{status: status, message: message}}
```

Comment:
In some ways, I prefer raising the errors directly. After all, the client code raises the error. If we treat the RPCs as plain functions, the errors should be raised.

Reply:
I was under the impression that elixir preferred to use this tuple construct versus raising errors? Failing RPCs are to be expected and I would not say that it's an exceptional thing. My main drive though was to be able to pattern match on the result of the call to be able to handle the error without diverting the execution flow with an exception.

Reply:
How about offering both? As in `recv!` and `recv` where the former raises and the latter returns a tuple?

Reply:
But for the unary rpc, we have to define two functions `xxx` and `xxx!`, which is annoying. Well, let's use the `{:error, }` tuple, just like what Erlang's rpc does.